mirrord/agent/src/main.rs

```diff
@@ -51,27 +52,87 @@ mod util;
 
 const CHANNEL_SIZE: usize = 1024;
 
+/// Keeps track of connected clients.
+/// If pausing target, also pauses and unpauses when number of clients changes from or to 0.
 #[derive(Debug)]
 struct State {
-    pub clients: HashSet<ClientID>,
+    clients: HashSet<ClientID>,
     index_allocator: IndexAllocator<ClientID>,
+    /// Was the pause argument passed? If true, will pause the container when no clients are
+    /// connected.
+    should_pause: bool,
```

Comment:
we should issue a warning that pausing without stealing might kill the original pod because of live checks.